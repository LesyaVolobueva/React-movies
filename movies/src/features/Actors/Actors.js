import React from 'react';
import { connect } from 'react-redux';
import { getActor } from '../MovieList/selectors';
import styles from './actors.module.scss';
import Nav from '../Nav/Nav';
import Header from '../Header/Header';
import PropTypes from 'prop-types';

const mapStateToProps = (state, ownProps) => ({
  actor: getActor(state, ownProps),
});

const Actor = (props) => {
  const { actor } = props;

  return (
    <div>
      <Nav />
      <Header />
      <div className={styles.actor}>
        <h2 className={styles.name}>{actor.name}</h2>
        <div className={styles.poster}>
          <img
            alt='poster'
            src={actor.imgUrl}
          />
        </div>
        <div>
          <span className={styles.bio}>Biography: </span>
          {actor.biography}
        </div>
      </div>
    </div>
  );
};

Actor.propTypes = {
  actor: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    imgUrl: PropTypes.string,
    biography: PropTypes.string,
  }),
};

export default connect(
  mapStateToProps,
  null,
)(Actor);
