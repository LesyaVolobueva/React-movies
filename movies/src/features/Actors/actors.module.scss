.actor {
  font-family: sans-serif;
  display: flex;
  flex-wrap: wrap;
  font-size: 16px;
  width: 50%;
  margin: 10px auto;
  box-shadow: 0 0 20px #a6a6a6;
  padding: 30px;

  & > div {
    flex-basis: 100%;
    margin-bottom: 15px;
  }
}

.name {
  font-size: 22px;
  font-weight: bold;
  text-align: center;
  flex-basis: 100%;
}

.poster {
  text-align: center;
  margin: 20px 0;

  img {
    width: 50%;
    height: auto;
  }
}

.bio {
  font-weight: bold;
}
