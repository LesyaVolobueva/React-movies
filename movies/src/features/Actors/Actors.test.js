import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Actors from './Actors';

jest.mock('../Nav/Nav', () => 'Nav');
jest.mock('../Header/Header', () => 'Header');

const actor = {
  "id": 28,
  "name": "Meat Loaf",
  "imgUrl": "https://images-na.ssl-images-amazon.com/images/M/MV5BMTIzNTQ4MjYyOV5BMl5BanBnXkFtZTcwNzgwNTEzMg@@._V1_UY317_CR6,0,214,317_AL_.jpg",
  "biography": "Lorem"
};

const lang = {
  edit: 'Edit',
  delete: 'Delete',
  save: 'Save',
  back: 'Go Back',
  signUp: 'Sign Up',
  login: 'Log In',
  movies: 'Movies',
  logOut: 'Log Out',
  imgUrl: 'Img Url',
  title: 'Title',
  director: 'Director',
  genres: 'Genres',
  description: 'Description',
  likes: 'Likes',
  actors: 'Actors',
  searchPlaceholder: 'Enter the name',
  byLikes: 'By likes',
  byRate: 'By rate',
  pleaseLogin: 'Please log in',
  userLogin: 'Your login',
  userPassword: 'Your password',
  passwordIncorrect: 'Password is incorrect!',
  userDoNotExist: 'User doesn\'t exist',
  dontHaveAcc: 'Don\'t have an account?',
  pleaseSignUp: 'Please sign up',
  suchUserExists: 'User with such login exists!',
  alreadyHaveAcc: 'Already have the account?',
};

const match = {
  params: {
    id: '28',
  },
};


describe('-- Actors --', () => {
  it('renders correctly', () => {

    const mockStore = configureStore();
    const store = {
      lang: {
        lang: 'en',
      },
      movies: {
        actors: [actor],
      },
    };

    const props = {
      match,
    };


    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={mockStore(store)}>
            <Actors {...props}/>
          </Provider>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
