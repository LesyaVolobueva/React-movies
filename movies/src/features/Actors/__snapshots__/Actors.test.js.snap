// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-- Actors -- renders correctly 1`] = `
<div>
  <Nav />
  <Header />
  <div
    className={undefined}
  >
    <h2
      className={undefined}
    >
      Meat Loaf
    </h2>
    <div
      className={undefined}
    >
      <img
        alt="poster"
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTIzNTQ4MjYyOV5BMl5BanBnXkFtZTcwNzgwNTEzMg@@._V1_UY317_CR6,0,214,317_AL_.jpg"
      />
    </div>
    <div>
      <span
        className={undefined}
      >
        Biography: 
      </span>
      Lorem
    </div>
  </div>
</div>
`;
